freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -358,14 +349,65 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
                 LOGGER.debug("Finished {}.", LogEvent.SYNC_CLOUD_IDENTITIES);
             }
 
+            if (fullSync && !warnings.isEmpty() && fmsToFreeipaBatchCallEnabled) {
+                UsersStateDifference usersStateDifferenceAfterSync = compareUmsAndFreeIpa(umsUsersState, fullSync, freeIpaClient);
+                if (usersStateDifferenceChanged(usersStateDifferenceBeforeSync, usersStateDifferenceAfterSync)) {
+                    LOGGER.info(String.format("Sync was partially successful for %s, thus we are trying it once again", stack.getResourceCrn()));
+                    warnings.clear();
+                    applyDifference(umsUsersState, environmentCrn, warnings, usersStateDifferenceAfterSync, freeIpaClient, fmsToFreeipaBatchCallEnabled);
+                }
+            }
```

Comment:
would be nice to move into a method and give it some nice name. eg. `retrySyncIfBatchCallHasWarnings`
I had to look at the log line to understand why we need this

Reply:
sure, i'll do it